docs/import-from-github.mdx

```diff
@@ -1,54 +1,80 @@
+2. Select the branch. You can choose any branch in your repo.
+3. Choose the files you want to sync. You can sync any JavaScript or TypeScript files, but only XState machines will be imported.
 
-[Watch Anders demo Import from Github during our office hours](https://www.youtube.com/watch?v=k-ymwkyUBao&t=365s).
+### Auto-sync
 
-:::
+Enabling **auto-sync** will automatically fetch the latest changes from your connected repository every time you open or refresh the project in the Studio.
+
+Disable auto-sync from the GitHub options from the <Github size={18}/> repository name in the footer of the left drawer.
+
+### Path for new files
+
+You can choose a **Path for new files** for when you create new machines in this connected project. These machines will be added at this relative path from the root of your repo. The default path is **src/stately-studio** and will be created in your repo when you create your first pull request in the Studio.
+
+### Create pull request
+
+Open the GitHub options from the <Github size={18}/> repository name in the footer of the left drawer. You can create pull requests from the GitHub options for any changes made to your machine.
+
+#### Update pull request
 
-How to import all machines from a GitHub repository into a project:
+If you have already created a pull request for your machine, you can update the pull request with any changes you have made in the Studio. Open the GitHub options from the <Github size={18}/> repository name in the footer of the left drawer and choose **Update pull request**.
 
-1. Navigate to **My Projects** from the sidebar or the Stately menu.
-2. Use **Import from GitHub** to open the Import repo from GitHub modal.
-3. The GitHub integration will fetch all available repositories (public and private). The search bar helps you filter the available repositories.
-4. Choose the repository from which you wish to import machines.
-5. Stately Studio will import your machines into a new private project using the same name as your repository and open your project in the editor.
+### Sync multiple branches
 
-You can [change the project’s visibility from the editor’s share dialog](projects.mdx#change-a-projects-visibility).
+You can sync multiple branches in your GitHub repo to the same project in Stately Studio. Open the GitHub branch options from the branch name in the footer of the left drawer to choose additional branches.
 
-### Import from a specific branch
+### Sync new machines created in Stately Studio
 
-Currently, the **Import from GitHub** feature imports all machines from the _default branch_ in your GitHub repository.
+Creating a new machine in Stately Studio in a connected project will flag the machine in your Machines list as **Not synced with GitHub**. You can sync the machine with your GitHub repo by [creating a pull request from the GitHub options](#create-pull-request) from the <Github size={18}/> repository name in the footer of the left drawer.
```

Comment:
Here's a thought that's not exclusive to this paragraph but it would probably help to refer to the modal UI that is opened from "repository name in the footer of the left drawer" as "GitHub Settings". That's what the menu item is named in Editor Menu > Project > GitHub Settings where it opens the same modal. If we start to use that terminology it'll be easier to document its contents from now on. We don't want to be stuck having to refer to the repo name in the footer of the left drawer if we can train users to think "GitHub Settings" after they learn how to open that from both locations (footer and Editor Menu).

Reply:
@kevinmaes I’m wary of using names when they’re not visible in the UI. Right now there’s no button that says “GitHub settings”, so as frustratingly verbose as the explanation is, it tells them where to find it.